nickfox_hw4_final.Rmd

```diff
@@ -0,0 +1,166 @@
+
+The data and [documentation](http://socserv.mcmaster.ca/jfox/Books/Applied-Regression-2E/datasets/Greene.pdf) were made avaialble by John Fox.  I have stored a copy of the data in Greene.txt in the subfolder data.
+
+
+```{r}
+suppressPackageStartupMessages(require(dplyr))
+suppressPackageStartupMessages(require(ggplot2))
+suppressPackageStartupMessages(require(broom))
+
+data <- read.table("data/Greene.txt")
+```
+
+Before doing anything else, just play with the data to get a sense of it.
+```{r}
+data %>%
+  group_by(decision) %>%
+  summarise(mean_decision = mean(decision))
+# "no" is dummy coded as 1, "yes" is dummy coded as 2
+
+data %>%
+  group_by(decision) %>%
+  summarise(n_decision = n())
+#270 no, 114 yes
+
+data %>%
+  group_by(language) %>%
+summarise(n_language = n())
+#253 English, 131 French
+
+```
+
+(@) The main outcome of interest is *decision* which records whether the judge granted an appeal or not.  In what proprotion of cases did the judge grant an appeal.
+
+```{r}
+
+decision <- data %>%
+  group_by(decision) %>%
+  summarise(n_decision = n()) %>%
+  mutate(prop_decision = (n_decision / sum(n_decision) * 100))
+
+print(decision)
+
+# Judges granted an appeal in 29.69% of cases.
+```
+
+(@) There are 12 (actually 10) different judges in the data. A key question is whether different judges have different rates of granting an appeal.  Make a plot showing the number of cases per judge.
+
+```{r}
+data %>%
+  group_by(judge) %>%
+  summarise(judge_count = n_distinct(judge))
+
+data_case <- data %>%
+  group_by(judge) %>%
+  summarise(judge_case = n()) %>%
+  arrange(desc(judge_case))
+
+print(data_case)
+
+data %>%
+  ggplot(aes(x = judge)) + geom_bar()
+
+```
+
+(@) Now plot the rate at which the judges grant appeals. 
+
+```{r}
+decision_judge <- data %>%
+  group_by(judge, decision) %>%
+  summarise(n_decision = n()) %>%
+  mutate(prop_yes = (n_decision / sum(n_decision) * 100)) %>%
+  filter(decision == "yes")
+
+decision_judge %>%
+  ggplot(aes(x = judge, y = prop_yes)) + geom_bar(stat = "identity")
+```
+
+(@) Now let's try this with logistic regression because we are going to move to more complicated models.  Fit a logistic regression predicting whether an appeal will be granted using judge as a categorical predictor. Use Iacobucci as the reference judge to facilitate the comparisons that follow. For more on how to control the reference level of a factor, check out this post on StackOverflow: http://stackoverflow.com/questions/3872070/how-to-force-r-to-use-a-specified-factor-level-as-reference-in-a-regression
+
+```{r}
+data <- within(data, judge <- relevel(judge, ref = "Iacobucci"))
+
+mylogit1 <- glm(decision ~ judge, data = data, family = "binomial")
+
+summary(mylogit1)
+
+```
+
+(@) Using the results of the model calculate the predicted probability that Judge Marceau will grant an appeal.
+
+```{r}
+estimates <- coef(mylogit1)
+
+prob_Marceau <- 1 / (1 + exp( - (estimates[1] + estimates[7] * 1)))
```

Comment:
Try to call the estimates with reference to the judge names instead of numbers because it is more robust and less prone to error. See solutions.